### PR TITLE
Support multiple HTTP inputs

### PR DESCRIPTION
**Pull request by gunnaraasen**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Adds support for multiple HTTP listeners. This is the last issue to completely fix https://github.com/influxdata/influxdb/issues/2785.

### CODE REVIEW

**Comment by gunnaraasen** [comment]
Looks like AppVeyor failed on an `influx_stress` test.

**Comment by e-dard** [comment]
LGTM 👍

**Comment by jwilder** [comment]
This is looks like a breaking change to the config file.  Are there any other inputs that are limited to only one?  

It would be nice to not have the config break.  Can we keep the existing HTTP config option and add it to the slice for backwards compatibility?

**Comment by gunnaraasen** [comment]
@jwilder I'm planning to open another PR to do exactly that for all the listeners but haven't finished yet.

**Comment by jwilder** [comment]
@gunnaraasen What happens when you use an old config that still uses `[http]` and not `[[http]]`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - [#5707](https://github.com/influxdata/influxdb/issues/5707): Return a deprecated message when IF NOT EXISTS is used.
 - [#6334](https://github.com/influxdata/influxdb/pull/6334): Allow environment variables to be set per input type.
 - [#6394](https://github.com/influxdata/influxdb/pull/6394): Allow time math with integer timestamps.
+- [#6333](https://github.com/influxdata/influxdb/pull/6333): Support for multiple listeners for HTTP inputs.
```
Comment (jwilder) [inline CHANGELOG.md]: Needs to be moved to next release.

File: cmd/influxd/run/server.go [hunk 2]
```diff
@@ -131,19 +131,26 @@ func NewServer(c *Config, buildInfo *BuildInfo) (*Server, error) {
...
 		reportingDisabled: c.ReportingDisabled,
+		tcpAddr:           bind,
+		config:            c,
+	}
 
-		httpAPIAddr: c.HTTPD.BindAddress,
-		httpUseTLS:  c.HTTPD.HTTPSEnabled,
-		tcpAddr:     bind,
+	for _, i := range c.HTTPDInputs {
+		if i.BindAddress != "" {
```
Comment (jwilder) [inline cmd/influxd/run/server.go]: So first listed HTTP input is the one we use for any code that returns an HTTP address?

**Comment by jwilder** [comment]
This needs to be updated to support `[http]` and log a deprecation warning similar to how we handled `[collectd]` and `[opentsdb]`.  See: https://github.com/influxdata/influxdb/blob/master/cmd/influxd/run/config.go#L133

**Comment by jsternberg** [comment]
What's the rationale behind this change? Unlike the other configuration changes, I don't see much value to multiple HTTP listeners. There are some downsides though with the naming behind the environment variable overrides and the configuration file change. If someone needs multiple HTTP listeners, what can we do better that Nginx can't do? Otherwise, I think we should just advise people to install a proper web server without risking the breaking change.

**Comment by jwilder** [comment]
I agree w/ @jsternberg.  I don't see the benefit of supporting this.

**Comment by gunnaraasen** [comment]
Mainly, the benefit would be the ability to set up an unauthenticated localhost listener for administrative access. It would also make this service consistent with the Graphite, OpenTSDB, UDP, and CollectD services.

I'm going to close this since it's a large change not necessary for 1.0.

**Pull request closed without merging**